staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go

```diff
@@ -202,7 +202,7 @@ func buildVersionSpecs(crd *apiextensionsv1.CustomResourceDefinition, oldSpecs m
 		if !v.Served {
 			continue
 		}
-		spec, err := builder.BuildSwagger(crd, v.Name, builder.Options{V2: true, StripDefaults: true})
+		spec, err := builder.BuildSwagger(crd, v.Name, builder.Options{V2: true, StripDefaults: false})
```

Comment:
where are these defaults stripped? This controller is the one creating the spec for later publishing.

Reply:
Is it https://github.com/kubernetes/kubernetes/pull/96480/commits/fb7a94bc740997452b8eb64fc765ff8b4decba40#diff-53cefa59736af8bf2b6669d4041bc7c6dacc08400f32e2348fcf42d0b0b180a5R193 in the aggregator that takes over the pruning? Maybe add a comment here.

Reply:
Yeah, we're pruning before serving, and keeping all the "internal representation" with the defaults.

Reply:
I've added a comment.